### PR TITLE
highfive: add v2.10.0

### PR DESCRIPTION
**Pull request by 1uc**
https://github.com/BlueBrain/HighFive/releases/tag/v2.10.0

### CODE REVIEW

**Review by alecbcs — APPROVED** [review]
Verified Checksums. Looks good to me. Waiting on final approval from @alkino.

**Review by alkino — APPROVED** [review]

**Pull request merged**
